pkg/csimapping/csimapping.go

```diff
@@ -0,0 +1,166 @@
+	keys := make(map[string]bool)
+	list := []string{}
+
+	for _, entry := range stringSlice {
+		if _, value := keys[entry]; !value {
+			keys[entry] = true
+			list = append(list, entry)
+		}
+	}
+	return list
+}
+func ListOperator(mapOperators map[OperatorKey][]string) []string {
+	keys := make(map[string]bool)
+	for key := range mapOperators {
+		keys[key.OperatorName] = true
+	}
+	var list []string
+	for key := range keys {
+		list = append(list, key)
+	}
+	return list
+}
+
+func getDriverNames() map[string]string {
+	out, err := exec.Command(driverNamesCommand).Output()
+	if err != nil {
+		log.Errorf("Command execution failed err:%v", err)
+	}
+
+	driverMap := make(map[string]string)
+
+	for _, line := range strings.Split(strings.TrimSuffix(string(out), "\n"), "\n") {
+		section := strings.Split(line, ",")
+		driverMap[section[1]] = section[0]
+	}
+	return driverMap
+}
+
+func createMapping(driverMap map[string]string, operatorList []string) map[string]string {
+	driverOperatorMapping := make(map[string]string)
+	for _, operator := range operatorList {
+		cleanedName := strings.ReplaceAll(operator, "-csi", "")
+		cleanedName = strings.ReplaceAll(cleanedName, "-certified", "")
+		cleanedName = strings.ReplaceAll(cleanedName, "-operator", "")
+		cleanedName = strings.ReplaceAll(cleanedName, "-stable", "")
+		cleanedName = strings.ReplaceAll(cleanedName, "k8s-", "")
+		cleanedName = strings.ReplaceAll(cleanedName, "bundle-", "")
+		cleanedName = strings.ReplaceAll(cleanedName, "-bundle", "")
+		cleanedName = strings.ReplaceAll(cleanedName, "csioperator", "")
+		cleanedName = strings.ReplaceAll(cleanedName, "-cluster", "")
+		for key, driverMeta := range driverMap {
+			matched, err := regexp.Match(cleanedName, []byte(driverMeta))
+			if err == nil && matched {
+				driverOperatorMapping[key] = operator
+			}
+		}
+	}
+	return driverOperatorMapping
+}
+
+func GetOperatorVersions() map[OperatorKey][]string {
+	var fullCatalog catalog
+	firstPageCatalog := getCatalogPage("https://catalog.redhat.com/api/containers/v1/operators/bundles?", 0, filterCsi)
+	totalPages := firstPageCatalog.Total / firstPageCatalog.PageSize
+	for i := uint(0); i < totalPages+1; i++ {
+		aCatalog := getCatalogPage("https://catalog.redhat.com/api/containers/v1/operators/bundles?", i, filterCsi)
+		fullCatalog.Data = append(fullCatalog.Data, aCatalog.Data...)
+		log.Debug(i)
+	}
+
+	mapOperators := make(map[OperatorKey][]string)
+	for _, entry := range fullCatalog.Data {
+		operatorName := strings.Split(entry.CsvName, ".")[0]
+		version := strings.Split(entry.CsvName, operatorName+".")[1]
+		aKey := OperatorKey{OperatorName: operatorName, OcpVersion: entry.OcpVersion}
+		aList := mapOperators[aKey]
+		aList = append(aList, version)
+		mapOperators[aKey] = aList
+	}
+	for key, operator := range mapOperators {
+		mapOperators[key] = removeDuplicateValues(operator)
+	}
+	return mapOperators
+}
+
+//nolint:deadcode // ok
+func main() {
+	mapOperators := GetOperatorVersions()
```

Comment:
Add a new make target to call this main code. Copy the csi-mapping.json to the final location at pkg/csimapping/csi-mapping.json